python/google/protobuf/internal/decoder.py

```diff
@@ -85,18 +85,17 @@
 __author__ = 'kenton@google.com (Kenton Varda)'
 
 import struct
-import sys  ##PY25
-_PY2 = sys.version_info[0] < 3  ##PY25
 from google.protobuf.internal import encoder
 from google.protobuf.internal import wire_format
+from google.protobuf.internal.utils import bytestr_to_string, b
 from google.protobuf import message
 
 
 # This will overflow and thus become IEEE-754 "infinity".  We would use
 # "float('inf')" but it doesn't work on Windows pre-Python-2.6.
-_POS_INF = 1e10000
-_NEG_INF = -_POS_INF
-_NAN = _POS_INF * 0
+_POS_INF = float('inf')
+_NEG_INF = float('-inf')
+_NAN = float('nan')
```

Comment:
@tseaver You make a good point. I'm working with the AppEngine owners to see if they could just freeze their Python 2.5 runtime at the current version of protobuf. This would allow us to drop 2.5 and possibly 2.6 support in this repo.

The downside is that new features (like maps, JSON suport, etc) would not be available for Python 2.5. But I think this might be a necessary tradeoff. Maintaining 2.5 support in this repo has been an increasing burden.